eth/downloader/downloader.go

```diff
@@ -1347,6 +1359,12 @@ func (d *Downloader) fetchParts(deliveryCh chan dataPack, deliver func(dataPack)
 			idles, total := idle()
 
 			for _, peer := range idles {
+				select {
```

Comment:
This is inconsistent. Why not to just check cancelCh, or perhaps just not to anything here and rely on fetch function and the outer loop to do the job